packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -9,10 +9,15 @@ const imagemin = require(`imagemin`)
 const imageminPngquant = require(`imagemin-pngquant`)
 const queue = require(`async/queue`)
 const path = require(`path`)
+const potrace = require(`potrace`)
```

Comment:
Let's lazy load these so they're only read off disk if someone is using them.

Reply:
Struggling! 👋 😄 Happy to explain in detail what I'm trying to do, but maybe this is enough to makes sense already: How do I find out if the query contains the `tracedSVG` field?

Reply:
Just move the requires inside the function for generating the traces.

Reply:
Yep! My problem is determining if the traced SVG should be generated at all – if I set `defaultValue` for the "tracedSVG" field, the traced SVG will always be generated; if I don't, tracedSVG will only be returned if one or more "trace" arguments are set. Or am I 🤒 again?

Reply:
Let me try and explain my issue again (still 😷🤒) – let's see if this amounts to anything more understandable then the previous attempt:

We want to lazy load potrace/SVGO only if needed. In the case of `duotone`, this would be easy – if there are duotone options set, apply the duotone effect – the image is always generated, anyway, and duotone options are required to generate a duotone image.

In the case of the "tracedSVG" however, I can't make the decision based on the "tracedSVG" options (`trace:{color…}`) being set or not, because that would require the user to always set options. Ideally, I could make the decision based on wether the "tracedSVG" is present in the query at all.

Reply:
Ahhh gotcha. Yeah so what you want to do here is add a resolve function to the actual field which is where you do the require and generate the svg. A field resolve function is only called if that field is actually queried so is the perfect place to put expensive work.

Reply:
😬😊🙏